OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessorLexer.cs

```diff
@@ -83,14 +83,30 @@ class DMPreprocessorLexer : TextLexer {
 
                         textBuilder.Append('@');
                         textBuilder.Append(delimiter);
-                        do {
-                            c = Advance();
 
-                            textBuilder.Append(c);
-                        } while (c != delimiter && c != '\n');
-                        Advance();
+                            /*
+                            do {
+                                c = Advance();
 
-                        string text = textBuilder.ToString();
+                                textBuilder.Append(c);
+                            } while (c != delimiter && c != '\n');
+                            Advance();
+                            */
+
+                            var everybody_gets_one = false;
+                            do {
+                                c = Advance();
+
+                                if (c == '\n' && !everybody_gets_one) {
+                                    everybody_gets_one = true;
+                                    c = '\0';
+                                    continue;
+                                }
+                                textBuilder.Append(c);
+                            } while (c != delimiter && !AtEndOfSource);
+                            Advance();
```

Comment:
A raw string can't have any newlines in it, unless it's a long raw string (`@{" ... "}`)

Reply:
Reverted, currently the parser will stop at long raw strings